typescript/packages/lookslike-high-level/src/components/window-manager.ts

```diff
@@ -596,14 +596,44 @@ export class CommonWindowManager extends LitElement {
     const recipeMatch = matchRoute("/recipe/:recipeId", url);
     if (recipeMatch) {
       const recipeId = recipeMatch.params.recipeId;
-      syncRecipe(recipeId).then(() => {
+      syncRecipe(recipeId).then(async () => {
         const recipe = getRecipe(recipeId);
         if (recipe) {
           // Get data from URL query parameter if it exists
           const searchParams = new URLSearchParams(url.search);
           const encodedData = searchParams.get("data");
           let initialData = {};
 
+          const fulfillUrl =
+            typeof window !== "undefined"
+              ? window.location.protocol + "//" + window.location.host + "/api/ai/spell/fulfill"
+              : "//api/ai/spell/fulfill";
+
+          // Call AI to analyze input schema and provide suggestions
+          const response = await fetch(fulfillUrl, {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+              accepts: "application/json",
+            },
+            body: JSON.stringify({
+              schema: recipe.argumentSchema?.properties,
+              many: false,
+              prompt: "",
+              options: {
+                format: "json",
+                validate: true,
+                maxExamples: 25,
+              },
+            }),
+          });
+
+          if (response.ok) {
+            const aiResponse = await response.json();
+            initialData = aiResponse.result;
+            console.log("AI response:", aiResponse);
+          }
+
```

Comment:
The dumbest thing that will work, open to relocating this code.

Reply:
I think we will want to iterate on `run` so that the casting can happen at a single place.  

Perhaps that means both window manager and the sidebar links for run recipe can trigger a `run(recipe, intent)` instead of `run(recipe, data)` - and if run gets intent it means "yo, cast this"

or perhaps we have a separate `cast` that handles the need to go from intent to the docs we care about.  then we can have `cast` -> `run` if you only have intent, and just call `run` once you have the actual docs you are working with

regardless - this lgtm for now!